context/app/static/js/components/workspaces/NewWorkspaceDialog/useCreateWorkspaceForm.ts

```diff
@@ -68,15 +72,16 @@ function useCreateWorkspaceForm({ defaultName, defaultTemplate, defaultProtected
   }
 
   useEffect(() => {
-    if (defaultProtectedDatasets && defaultProtectedDatasets !== '') {
+    if (initialProtectedDatasets && initialProtectedDatasets !== '') {
       reset({
-        'workspace-name': defaultName ?? '',
-        'protected-datasets': defaultProtectedDatasets ?? '',
-        templates: [defaultTemplate ?? DEFAULT_TEMPLATE_KEY],
+        'workspace-name': checkedWorkspaceName,
+        'protected-datasets': checkedProtectedDatasets,
+        templates: checkedTemplates,
         workspaceJobTypeId: DEFAULT_JOB_TYPE,
       });
     }
-  }, [defaultProtectedDatasets, reset, defaultName, defaultTemplate]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why the eslint-disable? I think you need to put the other `checked*` variables into the dependency array.